protocol.go

```diff
@@ -30,6 +30,9 @@ type Conn struct {
 	ProxyHeaderPolicy Policy
 	Validate          Validator
 	readErr           error
+	ReadHeaderTimeout time.Duration
+	ReadDeadline      time.Time
+	WriteDeadline     time.Time
```

Comment:
Do we really need to export these last two attributes?

Reply:
Nope! I'll update and add some documentation. Just wanted to add this as some boilerplate to show you what I was thinking but I'll get it merge ready.